public/app/features/plugins/admin/state/actions.ts

```diff
@@ -18,16 +19,74 @@ import { STATE_PREFIX } from '../constants';
 import { mapLocalToCatalog, mergeLocalsAndRemotes, updatePanels } from '../helpers';
 import { CatalogPlugin, RemotePlugin, LocalPlugin } from '../types';
 
+// Fetches
 export const fetchAll = createAsyncThunk(`${STATE_PREFIX}/fetchAll`, async (_, thunkApi) => {
   try {
-    const { dispatch } = thunkApi;
-    const [localPlugins, pluginErrors, { payload: remotePlugins }] = await Promise.all([
-      getLocalPlugins(),
-      getPluginErrors(),
-      dispatch(fetchRemotePlugins()),
-    ]);
-
-    return mergeLocalsAndRemotes(localPlugins, remotePlugins, pluginErrors);
+    thunkApi.dispatch({ type: `${STATE_PREFIX}/fetchLocal/pending` });
+    thunkApi.dispatch({ type: `${STATE_PREFIX}/fetchRemote/pending` });
+
+    const local$ = from(getLocalPlugins());
+    const remote$ = from(getRemotePlugins());
+    const pluginErrors$ = from(getPluginErrors());
+
+    forkJoin({
+      local: local$,
+      remote: remote$,
+      pluginErrors: pluginErrors$,
+    })
+      .pipe(
+        // Fetching the list of plugins from GCOM is slow / errors out
+        timeout({
+          each: 500,
+          with: () => {
+            remote$
+              // The request to fetch remote plugins from GCOM failed
+              .pipe(
+                catchError((err) => {
+                  thunkApi.dispatch({ type: `${STATE_PREFIX}/fetchRemote/rejected` });
+                  return throwError(() => new Error('Failed fetching plugins from GCOM.'));
```

Comment:
```suggestion
                  return throwError(() => new Error('Failed to fetch plugins from catalog (default https://grafana.com/grafana/plugins).'));
```

as per slack conversation 😄

Reply:
Although I wonder if this should actually be `https://grafana.com/api/plugins`?

Reply:
Indeed and indeed :) Should be added to the branch now 🙏